src/annotator/anchoring/xpath.js

```diff
@@ -20,50 +20,113 @@ export function xpathFromNode(el, relativeRoot) {
 }
 
 /**
- * Finds an element node using an XPath relative to the document root.
+ * Return the `index`'th immediate child of `element` whose tag name is
+ * `nodeName` (case insensitive).
+ *
+ * @param {Element} element
+ * @param {string} nodeName
+ * @param {number} index
  */
-function nodeFromXPathFallback(xpath, root) {
-  const steps = xpath.substring(1).split('/');
-  let node = root;
-  steps.forEach(step => {
-    let [name, idx] = step.split('[');
-    idx = idx ? parseInt(idx.split(']')[0]) : 1;
-    node = findChild(node, name.toLowerCase(), idx);
-  });
-  return node;
+function nthChildOfType(element, nodeName, index) {
```

Comment:
The `nthChildOfType` function is very similar to the removed `findChild` function except that it explicitly ignores case and returns `null` rather than throwing an exception if the element is not found, which is more consistent with how DOM APIs like `document.querySelector` or `document.evaluate` work.